src/app/components/algorithms/implementation-view/dialogs/implementation-nisq-analyzer-qpu-selection-prioritization-dialog/implementation-nisq-analyzer-qpu-selection-prioritization-dialog.component.ts

```diff
@@ -0,0 +1,130 @@
+import { Component, Inject, OnInit } from '@angular/core';
+import {
+  AbstractControl,
+  FormBuilder,
+  FormControl,
+  FormGroup,
+  Validators,
+} from '@angular/forms';
+import {
+  MAT_DIALOG_DATA,
+  MatDialog,
+  MatDialogRef,
+} from '@angular/material/dialog';
+import { XmcdaCriteriaService } from 'api-nisq/services/xmcda-criteria.service';
+
+@Component({
+  selector:
+    'app-implementation-nisq-analyzer-qpu-selection-prioritization-dialog',
+  templateUrl:
+    './implementation-nisq-analyzer-qpu-selection-prioritization-dialog.component.html',
+  styleUrls: [
+    './implementation-nisq-analyzer-qpu-selection-prioritization-dialog.component.scss',
+  ],
+})
+export class ImplementationNisqAnalyzerQpuSelectionPrioritizationDialogComponent
+  implements OnInit {
+  prioritizationFrom: FormGroup;
+  criteriaNamesAndValues: Criterion[] = [];
+  inputChanged = false;
+
+  constructor(
+    public dialogRef: MatDialogRef<
+      ImplementationNisqAnalyzerQpuSelectionPrioritizationDialogComponent
+    >,
+    @Inject(MAT_DIALOG_DATA) public data: DialogData,
+    public dialog: MatDialog,
+    private formBuilder: FormBuilder,
+    private mcdaService: XmcdaCriteriaService
+  ) {}
+
+  get mcdaMethod(): AbstractControl | null {
+    return this.prioritizationFrom.get('mcdaMethod');
+  }
+
+  get criteriaAndValues(): AbstractControl | null {
+    return this.prioritizationFrom.get('criteriaAndValues');
+  }
+
+  ngOnInit(): void {
+    this.onMcdaMethodChanged('topsis');
+  }
+
+  onNoClick(): void {
+    this.dialogRef.close();
+  }
+
+  onMcdaMethodChanged(mcdaMethod: string): void {
+    this.criteriaNamesAndValues = [];
+    this.mcdaService
+      .getCriterionForMethod({ methodName: mcdaMethod })
+      .subscribe((criteriaList) => {
+        // then get for each criterion the name and value and store both in the map as value
+        criteriaList.mcdaCriterionList.forEach((criterion) => {
```

Comment:
each MCDA method has its own list of criteria values, we add them into the list when the user selects a mcda method such that the current criteria values related to the mcda method are presented

Reply:
Yes, but why don't you do one single request to get all of the data needed instead of many small requests? Meaning, why does the criteriaList not already contain all information here?

Reply:
currently the values of the criteria are only accessible via a GET on a specific criterion... I will make an issue for the backend